### PR TITLE
Dependencies updates + doc

### PR DESCRIPTION
**Pull request by chevdor**
This PR:
- bumps + update deps
- apply required few fixes (Metadata Utf8PathBuf -> PathBuf)
- add/fix a bit of doc
- fix linting

### CODE REVIEW

File: src/patch.rs [hunk 1]
```diff
@@ -290,14 +290,13 @@ fn add_patches_for_packages(
             .entry(&p.name)
             .or_insert(Item::Value(Value::InlineTable(Default::default())))
             .as_inline_table_mut()
-            .ok_or_else(|| format!("Patch entry for `{}` isn't an inline table!", p.name))?;
+            .ok_or(format!("Patch entry for `{}` isn't an inline table!", p.name))?;
```
Comment (bkchr) [inline src/patch.rs:293]: Please revert

File: src/patch.rs [hunk 2]
```diff
@@ -290,14 +290,13 @@ fn add_patches_for_packages(
...
-        let path = if p.manifest_path.ends_with("Cargo.toml") {
+        if p.manifest_path.ends_with("Cargo.toml") {
             p.manifest_path.pop();
-            p.manifest_path
-        } else {
-            p.manifest_path
-        };
+        }
+
+        let path: PathBuf = p.manifest_path.into();
```
Comment (bkchr) [inline src/patch.rs:299]:
```suggestion
        let path = p.manifest_path;
```

File: src/patch.rs [hunk 3]
```diff
@@ -273,15 +273,15 @@ fn add_patches_for_packages(
...
-        .ok_or_else(|| "Patch table isn't a toml table!")?;
+        .ok_or( "Patch table isn't a toml table!")?;
 
     patch_table.set_implicit(true);
 
     let patch_target_table = patch_table
         .entry(&patch_target.as_string())
         .or_insert(Item::Table(Default::default()))
         .as_table_mut()
-        .ok_or_else(|| "Patch target table isn't a toml table!")?;
```
Comment (bkchr) [inline src/patch.rs:284]: Revert

File: src/patch.rs [hunk 4]
```diff
@@ -273,15 +273,15 @@ fn add_patches_for_packages(
         .entry("patch")
         .or_insert(Item::Table(Default::default()))
         .as_table_mut()
-        .ok_or_else(|| "Patch table isn't a toml table!")?;
```
Comment (bkchr) [inline src/patch.rs:276]: Same

File: src/patch.rs (hunk 2 above)
Comment (chevdor) [reply src/patch.rs:299]: The compiler disagrees with your suggestion. We need the type to be known for the call before L304 to be OK with calling `.display(...)`.

**Comment by chevdor** [comment]
I added the version bump so I suggest merging this PR after #8

**Review by bkchr — APPROVED** [review]

**Pull request merged**